app/models/spree/gateway/authorize_net.rb

```diff
@@ -2,14 +2,18 @@ module Spree
   class Gateway::AuthorizeNet < Gateway
     preference :login, :string
     preference :password, :string
+    preference :server, :string, :default => "test"
 
     def provider_class
       ActiveMerchant::Billing::AuthorizeNetGateway
     end
 
     def options_with_test_preference
-      options_without_test_preference.merge(test: self.preferred_test_mode)
+      raise "You must set the 'server' preference in your payment method (Gateway::AuthorizeNet) to either 'live' or 'test'" if !['live','test'].include?(self.preferred_server)
+      # warning: 'test' parameter indicates live or test server; it DOES NOT indicate Authorize.net test-mode
+      options_without_test_preference.merge(:test => (self.preferred_server != "live") )
```

Comment:
Will this be setting the gateway to the "useless" test mode when using the test server?